service/application/src/main/java/com/solace/maas/ep/event/management/agent/subscriber/SolacePersistentMessageHandler.java

```diff
@@ -57,26 +60,28 @@ public void onMessage(InboundMessage inboundMessage) {
                 messageClass = Class.forName(mopMessageSubclass);
                 cachedJSONDecoders.put(mopMessageSubclass, messageClass);
             }
-            MessageProcessor processor = messageProcessorsByClassType.get(messageClass);
+            processor = messageProcessorsByClassType.get(messageClass);
             if (processor == null) {
                 throw new UnsupportedOperationException("Could not find message processor for message of class " + messageClass.getCanonicalName());
             }
             setupMDC(messageAsString, messageClass.getSimpleName());
             log.trace("onMessage: {}\n{}", messageClass, messageAsString);
-            processor.processMessage(processor.castToMessageClass(toMessage(messageAsString, messageClass)));
+            message = toMessage(messageAsString, messageClass);
+            processor.processMessage(processor.castToMessageClass(message));
 
         } catch (Exception e) {
-            log.error("Error while processing inbound message from queue for mopMessageSubclass: {}", mopMessageSubclass);
-            throw new IllegalArgumentException(e);
+            if (processor != null && message != null) {
+                log.error("Error while processing inbound message from queue for mopMessageSubclass: {}", mopMessageSubclass);
+                processor.onFailure(e, processor.castToMessageClass(message));
+            } else {
+                log.error("Unsupported message and/or processor encountered. Skipping processing", e);
+            }
+
```

Comment:
Discussed with @mynecker. We MUST not throw exception from the message receiver thread, rather we should handle the exception. I already implemented the `processor.onFailure` for ConfigPushMessageProcessor. We SHOULD do the same thing for scan

Reply:
`processor.onFailure(e, processor.castToMessageClass(message));` exceptions that might be thrown within the `onFailure(...)` method should be handled, too. Logging the exception should be ok